Updating Datamapping with streamlit tutorial (#1164)
Snowflake-Labs/sfquickstarts: site/sfguides/src/data_mapping_in_native_apps/data_mapping_in_native_apps.md

The first solution is not really what we want to be doing because the consumer will potentially have to create objects on their Snowflake instance just to satisfy the application's requirements, so this Quickstart will deliver the second solution.

> aside positive
> 
> **Note** - The following steps explain how to create the app using the Snowflake CLI, as this option presents as a faster, more straightforward way to build Native Apps. You can also accomplish it by manually creating the folder structure and executing the SQL commands in Snowsight, but this alternative is out of the scope for this Quickstart.

The files explained in the following steps, are already present in the project folder, and it is merely a deeper explanation of its contents.

<!-- ------------------------ -->

> ## Snowflake CLI First Steps

Reviewer: I suggest we add a short paragraph that links to the official Snowflake CLI docs as to how to set up connections as a prerequisite.

Duration: 7

The datamapping native app project resides in a Snowflake repository which contains several app examples for the user to learn its way into Snowflake Native Apps.

The instructions on cloning the project repository and its execution are given in the readme file of this github [link NEEDS UPDATE AFTER PR](https://github.com/snowflakedb/native-apps-examples).

### Clone instructions 

1. Clone the project repo in your local machine. Repo could be cloned running:
    ```sh

>     snow app init --template-repo git@github.com:snowflakedb/native-apps-examples.git --template datamapping-app

Reviewer: Let's use `git clone` instead as we don't need the template expansion that `snow app init` provides.

    ```
2. Replace the CSV filepath inside the **prepare_data.sh** file, line 28, like this:  
    ```sh

>     snow object stage copy /USER_PATH_HERE/IP2LOCATION-LITE-DB11.CSV @location_data_stage --database ip2location --schema ip2location

Reviewer: Use `snow stage copy` and latest version of Snowflake CLI as the `snow object stage` commands are going to be deprecated in the next release.

    ```

<!-- ------------------------ -->
## Building the Application
Duration: 7

The application itself has been broken down into three parts.

* The building of the application package on the provider and the sharing of the lookup database with the application. 
* The building of the application install script which contains the functionality to accept an IP address, look it up in the database we just shared with the application and write back enhanced data from the application. 
* Arguably (certainly for this Quickstart) the most important part which is the user interface written using Streamlit. This is where we will do the mappings.

To do the enhancement of the IP addresses we will use a dataset called DB11 from [IP2LOCATION](https://www.ip2location.com/database/ip2location). There is a free version of this database available [here](https://lite.ip2location.com/database/db11-ip-country-region-city-latitude-longitude-zipcode-timezone), which is the one we will use in this quickstart.  If you do not have an account with them already you will need to create one. Download the dataset as a CSV file so it is ready to import  into the provider account.

> The first thing you need to do is create a new database which will serve as the lookup database for the application

Reviewer: We should add a period to this sentence. Also, the below code block is doing much more than just creating a database; is this all part of the prepare_data.sh file that's been copied in?

## Snowflake CLI First Steps
Duration: 7

The Snowflake CLI is a tool that allow us to run simple and powerful commands to communicate directly to your Snowflake account. You can find more information and install instructions [here](https://docs.snowflake.com/en/developer-guide/snowflake-cli-v2/installation/installation)
The datamapping native app project resides in a Snowflake repository which contains several app examples for the user to learn its way into Snowflake Native Apps.

> The instructions on cloning the project repository and its execution are given in the readme file of this github [link NEEDS UPDATE AFTER PR](https://github.com/snowflakedb/native-apps-examples).

Reviewer: Let's keep this PR unmerged until we have completed tasks for the linked repo.

zip_code varchar(30),
time_zone varchar(8)
);

--Create a file format for the file
CREATE OR REPLACE FILE FORMAT LOCATION_CSV
SKIP_HEADER = 1
FIELD_OPTIONALLY_ENCLOSED_BY = '\"'
COMPRESSION = AUTO;

--create a stage so we can upload the file
CREATE STAGE LOCATION_DATA_STAGE
file_format = LOCATION_CSV;"

--Go ahead now and upload the file to the stage.  After the file is uploaded go back to

> your worksheet and copy the file into the table created earlier

Reviewer: I suggest you split the `sh` block into two and have this as normal paragraph text instead of inside of the sh block (also `--` doesn't work as a comment for SH blocks)

> snow object stage copy /USER_PATH_HERE/IP2LOCATION-LITE-DB11.CSV @location_data_stage --database ip2location --schema ip2location 

Reviewer: Remove `object`, as above

snow sql -q "copy into litedb11 from @location_data_stage;
SELECT COUNT(*) FROM LITEDB11;SELECT * FROM LITEDB11 LIMIT 10;" --database ip2location --schema ip2location
```

We now have the reference database setup in the provider account so are now ready to start building the application itself.

<!-- ------------------------ -->
## Provider Setup
Duration: 5

> aside positive
> 

> > **Note** - The code is here to illustrate how would you normally do the app creation using the manual steps, but this step is executed automatically by the CLI, for example, in a manual environment you will need to update your files to a stage, but that managed by the **`snow app run`** command.  In the of the following steps there will be instructions on how to execute it.

Reviewer: It's not clear to me that we don't need to create the stage / application package, but we _do_ need to do the reference_usage and sharing stuff. Was that already moved over to a package script?

### Finishing the Application

> All the pieces are in place for our application, we just created (manifest.yml, setup_script.sql and enricher_dash.py) and now we are going to actually deploy it in the next page, in a manual environment you would have to upload the files to the stage APPLICATION_STAGE and then run other commands from a SQL worksheet, but we are not covering that.  

Reviewer: ```suggestion
All the pieces are in place for our application, we just created (manifest.yml, setup_script.sql and enricher_dash.py) and now we are going to actually deploy it in the next step.  
```

I suggest that we mention the manual alternative in the next page, where we actually show the deployment.

Once you have finished with the application and want to clean up your environment you can execute the following script

```sh

> snow sql -q " DROP APPLICATION APPL_MAPPING;

Reviewer: Instead of using SQL to drop the application + package, we need to use `snow app teardown`.
